lib/workers/repository/package-files.ts

```diff
@@ -24,7 +26,72 @@ export class PackageFiles {
     this.data.clear();
   }
 
-  public static getDashboardMarkdown(config: RenovateConfig): string {
+  /**
+   * Truncates the detected dependencies' section until it fits the available space
+   * i.e. It has length smaller than maxLength.
+   * This does not mutate the original PackageFiles data
+   * Note:  setHeader=false is used for testing purposes only
+   *        Mainly for comparing the output of getTruncatedMarkdown and getDashboardMarkdown
+   * @param config
+   * @param maxLength
+   * @param setHeader
+   */
+  public static getTruncatedMarkdown(
+    config: RenovateConfig,
+    maxLength: number,
+    setHeader = true
+  ): string {
+    const note =
+      '> **Note**\n> Detected dependencies section has been truncated\n';
+    const title = `## Detected dependencies\n\n`;
+
+    // exclude header length from the available space
+    const maxHeaderLen = setHeader ? (title + note).length : 0;
+    const mdMaxLength = maxLength - maxHeaderLen;
+
+    let md: string;
+    let header = '';
+    let removed = false;
+    let truncated = false;
+    let restore: (() => void) | null = null;
+
+    do {
+      // shorten markdown until it fits
+      md = PackageFiles.getDashboardMarkdown(config, false);
+      if (md.length > mdMaxLength) {
+        // backup data
+        if (!restore) {
+          restore = this.backup();
+        }
+        // truncate data
+        removed = PackageFiles.pop();
+      }
+      if (removed) {
+        truncated = true; // used to set the truncation Note
+      }
+    } while (removed && md.length > mdMaxLength);
+
+    if (restore) {
+      restore();
+    } // restore original PackageFiles data
+
+    header += title;
+    header += truncated ? note : '';
+
+    return (setHeader ? header : '') + md;
+  }
+
+  /**
+   * Generates the "detected dependencies" markdown
+   * Note:  setTitle=false is used for testing purposes only
+   *        Mainly for comparing the output of getDashboardMarkdown and getTruncatedMarkdown
+   * @param config
+   * @param setTitle
+   */
+  public static getDashboardMarkdown(
```

Comment:
This is only used in this class. IMO I would rename `getTruncatedMarkdown` to `getDashboardMarkdown`, rename `getDashboardMarkdown` to `getDashboardMarkdownInternal` (or something else) and make it `private`.